src/poller.ts

```diff
@@ -0,0 +1,69 @@
+import type { Options } from './types'
+
+export default class Poller {
+  private timeoutId: number
+  private isRunning: boolean
+
+  constructor(
+    private fetchFlagsFn: () => Promise<any>,
+    private configurations: Options,
+    private pollInterval: number,
+    private maxAttempts: number
+  ) {}
+
+  public start(): void {
+    if (this.isPolling()) {
+      this.logDebug('Polling was requested but is already running - only one poller can run at a time.')
+      return
+    }
+
+    this.isRunning = true
+
+    // Don't start polling immediately as we have alreadu fetched flags on client initialization
+    window.setTimeout(() => this.poll(), this.configurations.pollingInterval)
+  }
+
+  private poll(): void {
+    this.attemptFetch().finally(() => {
+      this.timeoutId = window.setTimeout(() => this.poll(), this.configurations.pollingInterval)
+    })
+  }
+
+  private async attemptFetch(): Promise<void> {
+    for (let i = 1; i <= this.maxAttempts; i++) {
+      const error = await this.fetchFlagsFn()
+
+      if (!error) {
+        this.logDebug('Successfully polled for flag updates')
+        return
+      }
+
+      this.logDebug('Error when polling for flag updates', error)
```

Comment:
Should this not filter down to `console.error` instead of `console.debug`? It's arguable either way I suppose.

Reply:
Will switch to error now that you mention it